### PR TITLE
FABI PCB Version

### PR DESCRIPTION
**Pull request by ChrisVeigl**
Various updates for FABI PCB Version:

*) added PCB layout and schematics
*) firmware upates for Arduino Pro Micro (I2C-Display, Neopixel & buzzer support, basic version of BT commands)
*) 3d printed case
*) construction 

(main contribution by Andreas Fußthaler)

### CODE REVIEW

**Comment by benjaminaigner** [comment]
@fussthalerAndreas @ChrisVeigl 
THX for this big improvement!

A few cleanup issues should be fixed before we can merge:
* Please remove the Diodes_THT 3D models (I think we need max. 1 file; but best would be the official KiCAD library without a file in our repo)
* KiCAD .bak Files are backup files which represent the last saved state, so with GIT we don't need them
* The same: .FCStd1 Files

File: FabiWare/parser.cpp [hunk 1]
```diff
@@ -77,7 +90,9 @@ void parseCommand (char * cmdstr)
     } 
        
     if (cmd>-1)  performCommand(cmd,num,actpos,0);        
-    else   Serial.println('?');              
+    else if (!PCBversion) {
+      Serial.println('?');    // TBD: why was this removed in PCB Version?
```
Comment (klues) [inline FabiWare/parser.cpp:94]: looks like this still needs attention.

File: FabiWare/parser.cpp [hunk 2]
```diff
@@ -95,7 +110,10 @@ void parseByte (int newByte)  // parse an incoming commandbyte from serial inter
             break;
         case 2: 
                 if ((newByte==13) || (newByte==10))  // AT reply: "OK" 
-                {  Serial.println(F("OK"));  readstate=0; }
+                {  
+                  if(!PCBversion)
+                    Serial.println(F("OK"));  readstate=0;
```
Comment (klues) [inline FabiWare/parser.cpp:115]: why no "OK" in PCB version?

File: FabiWare/parser.cpp [hunk 3]
```diff
@@ -105,9 +123,6 @@ void parseByte (int newByte)  // parse an incoming commandbyte from serial inter
                   readstate=0; }
                 else if(cmdlen<MAX_CMDLEN-1) cmdstring[cmdlen++]=newByte; 
             break;   
-        default: err: Serial.println('?');readstate=0;
+        default: err: if(!PCBversion){ Serial.println('?');readstate=0; }  // TBD: why was this removed in PCB Version?
```
Comment (klues) [inline FabiWare/parser.cpp:126]: why not in in PCB version?

**Review by klues — COMMENTED** [review]
Thanks for the improvements! 👍 
I've added some comments, please review.

**Comment by benjaminaigner** [comment]
There are still some differences regarding the AT API:
* `AT` should return "OK" (is used for keep-alive for the WebGUI)
* `AT LA` should return current config, if no config is available (initial startup), the default config should be copied to EEPROM.
* `AT CDDE` (unsupported command) should return "?".

**Comment by ChrisVeigl** [comment]
Serial output is now in again,
most of the TBDs are removed.

@benjaminaigner: this is still in question: 
https://github.com/asterics/FABI/blob/fdc57a7fac2495865e68dc20b2925120a0c58a23/FabiWare/bluetooth.cpp#L283

**Comment by benjaminaigner** [comment]
@ChrisVeigl 

That's right, this is code for Teensyduino.
It does not make sense to implement a second ASCII to keycode table, so we should use Arduino's built-in table (as we did with the FLipMouse/Teensyduino).

In Keyboard.cpp, we should use `const uint8_t _asciimap[128] PROGMEM;`
Because non-printable characters are defined too, we don't need to subtract 0x20:
`pgm_read_byte(_asciimap + k);`

I will investigate and commit the necessary changes.

**Pull request merged**